devicetypes/smartthings/viconics-schneider-room-controller.src/viconics-schneider-room-controller.groovy

```diff
@@ -0,0 +1,550 @@
+				}
+			} else if (attributeInt == CUSTOM_THERMOSTAT_MODE) {
+				log.debug "CUSTOM THERMOSTAT MODE, descMap.value: ${descMap.value}"
+				eventMap.name = "thermostatMode"
+				eventMap.value = THERMOSTAT_MODE_MAP[descMap.value]
+				eventMap.data = [supportedThermostatModes: state.supportedThermostatModes]
+			} else if (attributeInt == CUSTOM_THERMOSTAT_OPERATING_STATE) {
+				log.debug "CUSTOM THERMOSTAT OPERATING STATE, descMap.value: ${descMap.value}"
+				eventMap.name = "thermostatOperatingState"
+				eventMap.value = THERMOSTAT_OPERATING_STATE_MAP[descMap.value]
+			} else {
+				log.debug "descMap.inspect(): ${descMap.inspect()}"
+			}
+		}
+	}
+	result << createEvent(eventMap)
+	//log.debug "Description ${description} parsed to ${result}"
+	return result
+}
+
+private sendEventToChild(childNumber, event) {
+	def child = childDevices?.find { getChildId(it.deviceNetworkId) == childNumber }
+
+	if (child) {
+		log.debug "Sending ${event.name} event to $child.displayName"
+		child?.sendEvent(event)
+	} else {
+		log.debug "Child device $childNumber not found!"
+	}
+}
+
+def setCoolingSetpoint(degrees) {
+	setSetpoint(degrees, COOLING_SETPOINT)
+}
+
+def setHeatingSetpoint(degrees) {
+	setSetpoint(degrees, HEATING_SETPOINT)
+}
+
+def setChildCoolingSetpoint(deviceNetworkId, degrees) {
+	log.debug "deviceNetworkId: ${deviceNetworkId} degrees: ${degrees}"
+	def switchId = getChildId(deviceNetworkId)
+	if (switchId != null) {
+		setSetpoint(degrees, COOLING_SETPOINT_UNOCCUPIED)
+	}
+}
+
+def setChildHeatingSetpoint(deviceNetworkId, degrees) {
+	log.debug "deviceNetworkId: ${deviceNetworkId} degrees: ${degrees}"
+
+	def switchId = getChildId(deviceNetworkId)
+	if (switchId != null) {
+		setSetpoint(degrees, HEATING_SETPOINT_UNOCCUPIED)
+	}
+}
+
+def getChildId(deviceNetworkId) {
+	def split = deviceNetworkId?.split(":")
+	return (split.length > 1) ? split[1] as Integer : null
+}
+
+def setSetpoint(degrees, setpointAttr) {
+	log.debug "degrees: ${degrees}, setpointAttr: ${setpointAttr}"
+	if (degrees != null && setpointAttr != null) {
+		log.debug "temperatureScale: ${temperatureScale}"
+		def celsius = (temperatureScale == "C") ? degrees : fahrenheitToCelsius(degrees)
+		celsius = (celsius as Double).round(2)
+
+		delayBetween([
+				zigbee.writeAttribute(THERMOSTAT_CLUSTER, setpointAttr, DataType.INT16, zigbee.convertToHexString(celsius * 100)),
+				zigbee.readAttribute(THERMOSTAT_CLUSTER, setpointAttr)
+		], 500)
+	}
+}
+
+def setThermostatFanMode(mode) {
+	if (state.supportedFanModes?.contains(mode)) {
+		if (isViconicsVT8350() || isSchneiderSE8350()) {
+			switch (mode) {
+				case "on":
+					setFanSpeed(1)
+					break
+				case "auto":
+					setFanSpeed(4)
+					break
+			}
+		} else if (isViconicsVT8650() || isSchneiderSE8650()) {
+			switch (mode) {
+				case "on":
+					getThermostatFanModeCommands(CUSTOM_FAN_MODE_ON)
```

Comment:
I think you should avoid a switch statement here and delegate this lookup of the fan mode value to the `getThermostatFaneModeCommands` function you defined.

Reply:
getThermostatFaneModeCommands receives constant hex values, but a map will be in the middle - that way switch statement won't be used here.